src/raftstore/store/peer_storage.rs

```diff
@@ -319,12 +319,47 @@ impl PeerStorage {
         DbSnapshot::new(self.engine.clone())
     }
 
+    fn validate_snap(&self, snap: &Snapshot) -> bool {
+        let idx = snap.get_metadata().get_index();
+        if idx < self.truncated_index() {
+            // stale snapshot, should generate again.
+            info!("{} snapshot {} < {} is stale, generate again.",
+                  self.tag,
+                  idx,
+                  self.truncated_index());
+            return false;
+        }
+
+        let mut snap_data = RaftSnapshotData::new();
+        if let Err(e) = snap_data.merge_from_bytes(snap.get_data()) {
+            error!("{} decode snapshot fail, it maybe corrupted: {:?}",
+                   self.tag,
+                   e);
+            return false;
+        }
+        let snap_epoch = snap_data.get_region().get_region_epoch();
+        let latest_epoch = self.get_region().get_region_epoch();
+        if snap_epoch.get_conf_ver() < latest_epoch.get_conf_ver() {
```

Comment:
Version check is unnecessary, because when applying a log entry we only assume the peer should be a part of the region. Conf version check ensures it perfectly.